src/Framework/FieldsAPI/FormField/FieldOptions.php

```diff
@@ -2,29 +2,107 @@
 
 namespace Give\Framework\FieldsAPI\FormField;
 
-use Give\Framework\FieldsAPI\FormField\FieldTypes;
+use Give\Helpers\Arr;
 
 trait FieldOptions {
 
-	/** @var array */
+	/** @var FieldOption[] */
 	protected $options = [];
 
+	/**
+	 * Does the field type support options?
+	 *
+	 * @return bool
+	 */
 	public function supportsOptions() {
 		return in_array(
 			$this->getType(),
 			[
 				FieldTypes::TYPE_SELECT,
 				FieldTypes::TYPE_RADIO,
-			]
+			],
+			true
 		);
 	}
 
-	public function options( $options ) {
-		$this->options = $options;
+	/**
+	 * Set the options
+	 *
+	 * @param array $options
+	 *
+	 * @return $this
+	 */
+	public function options( array $options ) {
+		// Reset options, since they are meant to be set immutably
+		$this->options = [];
+
+		// Determine if the provided options are an associative array before the loop to avoid extra work.
+		$areOptionsAnAssocArray = Arr::isAssoc( $options );
+
+		// Loop through the options and transform them to the proper format.
+		foreach ( $options as $maybeLabel => $fieldOptionOrArrayOrValue ) {
+			if ( $fieldOptionOrArrayOrValue instanceof FieldOption ) {
+				// Since the given format matches the proper format (i.e. an array of FieldOption),
+				// we can avoid additional cycles of the loop by just setting the value and breaking
+				// out of it early.
+				$this->options = $options;
+				break;
+			}
```

Comment:
Let's allow for a mixed bag. Just append this FieldOption and move on to the next.

Reply:
Resolved in 7d94367